src/MPO.jl

```diff
@@ -0,0 +1,141 @@
+using Random
+using Graphs
+using BijectiveDicts: BijectiveIdDict
+
+abstract type AbstractMPO <: AbstractAnsatz end
+
+"""
+    MPO <: AbstractAnsatz
+
+A Matrix Product Operator (MPO) [`Ansatz`](@ref) Tensor Network.
+"""
+struct MPO <: AbstractAnsatz
+    tn::Ansatz
+    form::Form
+end
+
+Ansatz(tn::MPO) = tn.tn
+
+Base.copy(x::MPO) = MPO(copy(Ansatz(x)), form(x))
+Base.similar(x::MPO) = MPO(similar(Ansatz(x)), form(x))
+Base.zero(x::MPO) = MPO(zero(Ansatz(x)), form(x))
+
+defaultorder(::Type{MPO}) = (:o, :i, :l, :r)
+boundary(::MPO) = Open()
+form(tn::MPO) = tn.form
+
```

Comment:
""" Builds an MPO from a list of input tensors. Default ordering for the legs is (physical out, physical in, virtual left, virtual right) - can be specified with the order keyword argument.
Does/Does not? check for consistent dimension of input tensor legs
"""